src/terrain/terrain.js

```diff
@@ -59,6 +62,50 @@ type RenderBatch = {
     end: number;
 }
 
+class MockSourceCache extends SourceCache {
+    mockDEM: DEMData;
+
+    constructor(map: Map) {
+        const sourceSpec = {type: 'raster-dem', maxzoom: map.transform.maxZoom};
+        const sourceDispatcher = new Dispatcher(getWorkerPool(), null);
+        const source = createSource('mock-dem', sourceSpec, sourceDispatcher, map.style);
+
+        super('mock-dem', source, false);
+
+        source.setEventedParent(this);
+
+        this.used = this._sourceLoaded = true;
+        this.mockDEM = this._createMockDEM();
+    }
+
+    _loadTile(tile: Tile, callback: Callback<void>) {
+        tile.dem = this.mockDEM;
```

Comment:
I might have missed here something, but why do we need to set a mock dem here? I've tried to set it to undefined and it was still working for me as before. Isn't it enough to set the source cache?

Reply:
Good point, that works and simplifies this a bit. Because we fallback to an empty DEM texture line 642 when this is set to undefined:
```js
const demTexture = this._prepareDemTileUniforms(tile, demTile, uniforms) ?
    (demTile.demTexture: any) : this.emptyDEMTexture;
    demTexture.bind(gl.NEAREST, gl.CLAMP_TO_EDGE);
```